chainer/function_hooks/cupy_memory_profile.py

```diff
@@ -129,10 +129,12 @@ def summary(self):
     def _choose_unit(self, size):
         """Choose optimal unit."""
         denomi = 1
+        if size <= 0:
+            return denomi, self._units[0]
         for unit in self._units[:-1]:
-            if size / denomi >= 1:
+            if size / (denomi * 1024.0) < 1:
                 return denomi, unit
-            denomi *= 1024.0
+            denomi *= 1024
```

Comment:
It seems `from __future__ import division` will be needed (L170--172).

Reply:
Thanks, changed to always return denomi as float, as it was the original spec.